### PR TITLE
Shadow Walk/New Shadowling movement spell

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Added Shadow Walk, a new form of jaunt that lasts for an unlimited amount of time, but ends if you enter a well lit tile. It is planned for use in a Shadowling rework, but feel free to hassle admins to test it out in the mean time.
/:cl:

Can only be used in the dark/will end if you leave the dark.

It also heals you while you're in the darkness.

Shadowlings for side antag 2017. This will make expanding areas of light/dark actually relevant

### CODE REVIEW

File: code/modules/spells/spell_types/shadow_walk.dm [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+/obj/effect/dummy/shadow/relaymove(mob/user, direction)
+	forceMove(get_step(src,direction))
+	check_light_level()
+
+/obj/effect/dummy/shadow/proc/check_light_level()
+	var/turf/T = get_turf(src)
+	var/light_amount = T.get_lumcount()
+	if(light_amount > 0.2) // jaunt ends
+		end_jaunt(TRUE)
+	else if (light_amount < 0.2 && jaunter) //heal in the dark
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/shadow_walk.dm]: probably only needs to be an if(!QDELETED(jaunter))

File: code/modules/spells/spell_types/shadow_walk.dm [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	charge_max = 0
+	clothes_req = 0
+	phase_allowed = 1
+	selection_type = "range"
+	range = -1
+	include_user = 1
+	cooldown_min = 0
+	overlay = null
+	action_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	action_icon_state = "bloodcrawl"
```
Comment (KorPhaeron) [inline code/modules/spells/spell_types/shadow_walk.dm]: Any suggestions for what icon to use?

**Comment by JamieH** [comment]
a

**Comment by Armhulen** [comment]
Just give it to shadowpeople in the meantime

File: code/modules/spells/spell_types/shadow_walk.dm (hunk 2 above)
Comment (ShizCalev) [reply code/modules/spells/spell_types/shadow_walk.dm]: shadowwalk

try this for size

**Comment by Armhulen** [comment]
also, thanks kor i appreciate it!!

**Comment by Super3222** [comment]
https://github.com/yogstation13/yogstation/pull/75

**Comment by KorPhaeron** [comment]
@super3222

Didnt realize another server had tried this. How do you find it plays balance wise?

**Comment by Super3222** [comment]
It works well. You're going to want to make veil disable drone lights, almost permanently. Drones are huge shitters when it comes to this (even with their laws). Aside from that, I can't really say because I don't know /tg/ that well.

**Comment by MoreRobustThanYou** [comment]
ehhhh I prefer the passive ability version (like on yog)

**Comment by KorPhaeron** [comment]
Yog version touches a bunch of mob/movement code for the sake of one antag power, it would't be accepted here

**Comment by MoreRobustThanYou** [comment]
@KorPhaeron iirc, other things on yogs shadow walk, like statues

**Comment by KorPhaeron** [comment]
Weeping angels cant phase through walls that isn't canon

**Pull request merged**